cli/client/main.go

```diff
@@ -416,14 +413,33 @@ func mainWithStatusCode() int {
 	}
 
 	hostnameIsAnIP := net.ParseIP(hostname) != nil
-
-	port := configPort
-	if port == -1 && urlPort != "" {
-		port, err = strconv.Atoi(urlPort)
-		if err != nil {
-			log.Error().Msgf("invalid port number: %s: %s", urlPort, err)
+	// *** Port selection ***
+	// There are six scenarios:
+	//
+	// | CLI port | parse | config port | program behaviour    |
+	// | -------- | ----- | ----------- | -------------------- |
+	// |    yes   |  nok  |     yes     | throw error and exit |
+	// |    yes   |  nok  |     no      | throw error and exit |
+	// |    yes   |  ok   |     yes     | use CLI port         |
+	// |    yes   |  ok   |     no      | use CLI port         |
+	// |    no    |  ---  |     yes     | use config port      |
+	// |    no    |  ---  |     no      | use default port     |
+
+	var port int
+	if urlPort != "" {
+		if parsedPort, err := strconv.Atoi(urlPort); err != nil || parsedPort > 0xffff {
+			// use WithLevel(zerolog.FatalLevel) to log a fatal level, but let us handle
+			// program termination. log.Fatal() exits with os.Exit(1).
+			log.WithLevel(zerolog.FatalLevel).Str("Port", urlPort).Err(err).Msg("cli contains an invalid port")
+			fmt.Fprintf(os.Stderr, "Bad port '%s'\n", urlPort)
 			return -1
+		} else {
+			port = parsedPort
 		}
+	} else if configPort != -1 {
+		port = configPort
+	} else {
+		port = 443
```

Comment:
Add a comment, like `// no explicit port from the CLI and no port specified in config` ?